pkg/controllers/observability-ui/observability-ui-plugin/controller.go

```diff
@@ -0,0 +1,218 @@
+package observability_ui_plugin
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	osv1alpha1 "github.com/openshift/api/console/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metaerrors "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	obsui "github.com/rhobs/observability-operator/pkg/apis/observability-ui/v1alpha1"
+	"github.com/rhobs/observability-operator/pkg/reconciler"
+)
+
+type resourceManager struct {
+	k8sClient      client.Client
+	scheme         *runtime.Scheme
+	logger         logr.Logger
+	controller     controller.Controller
+	pluginConf     ObservabilityUIPluginsConfiguration
+	clusterVersion string
+}
+
+type ObservabilityUIPluginsConfiguration struct {
+	Images map[string]string
+}
+
+type Options struct {
+	PluginsConf ObservabilityUIPluginsConfiguration
+}
+
+// RBAC for managing ObservabilityUIPlugins
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins,verbs=get;list;watch;create;update;delete;patch
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins/status,verbs=get;update
+
+// RBAC for managing observability ui plugin objects
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=serviceaccounts;services;configmaps,verbs=get;list;watch;create;update;patch;delete
+
+// RBAC for managing Console CRs
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=consoles,verbs=get;patch;list;watch
+// +kubebuilder:rbac:groups=console.openshift.io,resources=consoleplugins,verbs=get;list;watch;create;update;delete;patch
+
+// RBAC for reading cluster version
+// +kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch
+
+func RegisterWithManager(mgr ctrl.Manager, opts Options) error {
+	logger := ctrl.Log.WithName("observability-ui")
+
+	clusterVersion, err := getClusterVersion(mgr.GetAPIReader())
```

Comment:
How does it work with cluster upgrades? We don't need to answer in this PR but maybe I expect that "something" watches the clusterversion object and reconciles the state when a change happens.

Reply:
Created a task to follow up on this: https://issues.redhat.com/browse/COO-106